ansible_ai_connect/ai/api/model_client/wca_client.py

```diff
@@ -514,6 +517,32 @@ def get_model_id(
         logger.error("Seated user's organization doesn't have default model ID set")
         raise WcaModelIdNotFound(model_id=requested_model_id)
 
+    def get_default_model_id(
+        self,
+        user,
+        organization_id: Optional[int] = None,
+        requested_model_id: str = "",
+    ):
+        model_id = self.get_model_id(user, organization_id, requested_model_id)
+
+        # If the model_id is found in the list of known models that is used for a specific API,
+        # send it to WCA playbook explanation/generation API may cause an issue. Especially it
+        # can occur the model_is is "granite-3b", which is used only for task generation.
+        #
+        # Following if block is for getting rid of such model_id before calling WCA playbook
+        # explanation/generation API.
+        separator = "<|sepofid|>"
+        index = model_id.find(separator)
+
+        if index > -1:
+            wca_model_id = model_id[(index + len(separator)) :]
```

Comment:
I think this should probably be `model_id[index:]`.

Reply:
Updated the code. Thanks!